freeze.sh

```diff
@@ -0,0 +1,51 @@
+#! /bin/bash
+
+set -e
+set -x
+
+source manifest
+
+# Allow multiple downloads
+sed -i '/ParallelDownloads/s/^#//g' /etc/pacman.conf
+# set archive date if specified and force update/downgrade builder
+if [ -n "${ARCHIVE_DATE}" ]; then
+	echo "
+	Server=https://archive.archlinux.org/repos/${ARCHIVE_DATE}/\$repo/os/\$arch
+	" > /etc/pacman.d/mirrorlist
+fi
+
+# Update system and install pikaur and other build dependencies
+echo -e "[multilib]\nInclude = /etc/pacman.d/mirrorlist\n" >> /etc/pacman.conf && \
+	pacman --noconfirm -Syyu && \
```

Comment:
This needs another `-u` to force downgrade of installed packages after setting `ARCHIVE_DATE` to avoid package drifting.